sky/backends/backend_utils.py

```diff
@@ -706,6 +706,15 @@ def write_cluster_config(
                 'is not supported by this cloud. Remove the config or set: '
                 '`remote_identity: LOCAL_CREDENTIALS`.')
         excluded_clouds = [cloud]
+
+    for cloud_str, cloud_obj in cloud_registry.CLOUD_REGISTRY.items():
+        remote_identity_config = skypilot_config.get_nested(
+            (cloud_str.lower(), 'remote_identity'), None)
+        if remote_identity_config:
+            if (remote_identity_config ==
+                    schemas.RemoteIdentityOptions.NO_UPLOAD.value):
+                excluded_clouds.append(cloud_obj)
+
```

Comment:
The error message raised above can be misleading if a user specifies `NO_UPLOAD` for the current cloud. Should we update that?

Reply:
Ah thanks, updated now